plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java

```diff
@@ -169,14 +169,18 @@ private ClientState buildClient(AzureStorageSettings azureStorageSettings, BiCon
         final NioEventLoopGroup eventLoopGroup = new NioEventLoopGroup(new NioThreadFactory());
         final NettyAsyncHttpClientBuilder clientBuilder = new NettyAsyncHttpClientBuilder().eventLoopGroup(eventLoopGroup);
 
-        final Proxy proxy = azureStorageSettings.getProxy();
-        if (proxy != null) {
-            final Type type = Arrays.stream(Type.values())
-                .filter(t -> t.toProxyType().equals(proxy.type()))
-                .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException("Unsupported proxy type: " + proxy.type()));
-
-            clientBuilder.proxy(new ProxyOptions(type, (InetSocketAddress) proxy.address()));
+        final ProxySettings proxySettings = azureStorageSettings.getProxySettings();
+        if (proxySettings != ProxySettings.NO_PROXY_SETTINGS) {
+            if (proxySettings.isAuthenticated()) {
+                Authenticator.setDefault(new Authenticator() {
+                    @Override
+                    protected PasswordAuthentication getPasswordAuthentication() {
+                        return new PasswordAuthentication(proxySettings.getUsername(), proxySettings.getPassword().toCharArray());
+                    }
+                });
+            }
+            final Type azureProxyType = proxySettings.getType() == Proxy.Type.SOCKS ? Type.SOCKS5 : Type.HTTP;
```

Comment:
I am not sure we should discard the possibility of `Type.SOCKS4` usage since we don't know all possible deployment scenarios. May be it makes sense to widen the proxy types (detach from enumeration) to `DIRECT`, `HTTP`, `SOCKS` (default to `SOCKS4` to support compatibility since this is what is being picked right now), `SOCKS4`, `SOCKS5`?

Reply:
That the problem that Azure SDK wraps up `Proxy.Tpe` and the real type for both - is Proxy.Type.SOCKS`. We can set any in this case, since JDK take scare about SOCKS version

Reply:
Now I get what do you mean. Let's use Azure SDK `ProxyType` explicitly

Reply:
done. I had had to add `ProxySettings.ProxyType`. It wraps Azure SDK `Type` and use NONE as a default value, since Settings cannot be null, and we need to check if ProxyType was set or not.

Reply:
:+1: